src/Core/src/Platform/iOS/WindowExtensions.cs

```diff
@@ -15,23 +16,50 @@ internal static void UpdateTitle(this UIWindow platformWindow, IWindow window)
 			// If you set it to an empty string the title reverts back to the 
 			// default app title.
 			if (OperatingSystem.IsIOSVersionAtLeast(13) && platformWindow.WindowScene is not null)
+			{
 				platformWindow.WindowScene.Title = window.Title ?? String.Empty;
+			}
 		}
 
 		internal static void UpdateX(this UIWindow platformWindow, IWindow window) =>
-			platformWindow.UpdateUnsupportedCoordinate(window);
+			platformWindow.UpdateCoordinates(window);
 
 		internal static void UpdateY(this UIWindow platformWindow, IWindow window) =>
-			platformWindow.UpdateUnsupportedCoordinate(window);
+			platformWindow.UpdateCoordinates(window);
 
 		internal static void UpdateWidth(this UIWindow platformWindow, IWindow window) =>
-			platformWindow.UpdateUnsupportedCoordinate(window);
+			platformWindow.UpdateCoordinates(window);
 
 		internal static void UpdateHeight(this UIWindow platformWindow, IWindow window) =>
-			platformWindow.UpdateUnsupportedCoordinate(window);
+			platformWindow.UpdateCoordinates(window);
 
-		internal static void UpdateUnsupportedCoordinate(this UIWindow platformWindow, IWindow window) =>
-			window.FrameChanged(platformWindow.Bounds.ToRectangle());
+		internal static void UpdateCoordinates(this UIWindow platformWindow, IWindow window)
+		{
+			var callFrameChanged = true;
+			var rectangle = platformWindow.Bounds.ToRectangle();
+
+			if (OperatingSystem.IsIOSVersionAtLeast(16)) 
+			{
+				var windowScene = platformWindow.WindowScene;
+
+				if (windowScene is not null)
+				{
+					UIWindowSceneGeometryPreferencesMac preferences = new()
+					{
+						SystemFrame = new CGRect(window.X, window.Y, window.Width, window.Height)
+					};
+
+					// TODO: Log errors.
+					windowScene.RequestGeometryUpdate(preferences, (error) => { });
+					callFrameChanged = false;
+				}
+			}
+
+			if (callFrameChanged) 
+			{
+				window.FrameChanged(rectangle);
+			}
```

Comment:
Can we collapse this logic into a single if/else block?

```suggestion
			if (OperatingSystem.IsIOSVersionAtLeast(16) && platformWindow.WindowScene is {} windowScene))
			{
				var preferences = new UIWindowSceneGeometryPreferencesMac()
				{
					SystemFrame = new CGRect(window.X, window.Y, window.Width, window.Height)
				};

				// TODO: Log errors.
				windowScene.RequestGeometryUpdate(preferences, (error) => { });
			}
			else
			{
				var rectangle = platformWindow.Bounds.ToRectangle();
				window.FrameChanged(rectangle);
			}
```